### PR TITLE
Follow the 'connections' uri when present.

### PR DESCRIPTION
**Pull request by bable5**
* Full access to the API requires both user consent and designating
which organizations the user wants the application to be able to
interact with. Follow the 'connections' uri if present on any org to
allow the user finish the setup process.
* Re-worked the `/` controller slightly. Previously it was always
updating the session `settings` with a new set of values from the oauth
configuration. This caused the access token in the session to be lost
when coming back to the application after the connections org-select
redirect. Now it only initializes session `settings` if it is not set.

### CODE REVIEW

**Review by jdoubleyou — APPROVED** [review]

File: app/routes.php [hunk 1]
```diff
@@ -84,28 +91,29 @@
...
         $token = filter_var($parsedBody['token'], FILTER_SANITIZE_STRING);
+
+        $body = apiGet($token, $url);
+        $body = json_encode(json_decode($body), JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES);
+
+        $settings['accessToken'] = $token;
+        $settings['apiResponse'] = $body;
+        $_SESSION['settings'] = $settings;
+
+        return$container->render($response, 'main.phtml', ["settings" => $settings]);
```
Comment (jdoubleyou) [inline app/routes.php:102]: I _feel_ bad for pointing this out....but you missed a space.

File: app/routes.php [hunk 2]
```diff
@@ -14,11 +14,11 @@
...
-        $settings = $oauth->getSettings();
-        $_SESSION['settings'] = $settings;
-        // $response->getBody()->write('To print on page');
-        $response = $container->render($response, 'main.phtml',  ["settings" => $settings]);
+        if (!isset($_SESSION['settings'])) {
+            $_SESSION['settings'] = (new OAuth2Settings())->getSettings();
+        }
+
+        $response = $container->render($response, 'main.phtml', ["settings" => $_SESSION['settings']]);
         return $response;
```
Comment (jdoubleyou) [inline app/routes.php:22]: Unneeded variable?

File: templates/main.phtml [hunk 3]
```diff
@@ -188,8 +131,8 @@
 
     </div>
 <?php } ?>
-<!--
-<link rel="stylesheet" href="{{ url_for('static', filename='functions.js') }}">
-<link rel="stylesheet" href="{{ url_for('static', filename='prism.js') }}">-->
+
+<script type="text/javascript" src="/static/functions.js"></script>
+<script type="text/javascript" src="/static/prism.js"></script>
```
Comment (jdoubleyou) [inline templates/main.phtml:136]: Why are we putting script tags at the bottom of the html? Isn't this a weird hack to get the page to load faster by not loading javascript until the last second? These should probably be in <head>

**Review by jdoubleyou — CHANGES_REQUESTED** [review]

File: templates/main.phtml (hunk 3 above)
Comment (bable5) [reply templates/main.phtml:136]: Because that's where they are in all the examples? At least it's consistent.

File: app/routes.php (hunk 1 above)
Comment (bable5) [reply app/routes.php]: :disappointed:

File: templates/main.phtml (hunk 3 above)
Comment (jdoubleyou) [reply templates/main.phtml:136]: fair enough

**Review by jdoubleyou — APPROVED** [review]

**Comment by jdoubleyou** [comment]
retest this please

**Comment by jdoubleyou** [comment]
was worth a shot

**Review by jdoubleyou — APPROVED** [review]

**Pull request merged**